### PR TITLE
WIP add XYZ format and QM9 example

### PR DESCRIPTION
**Pull request by streeve**
- No longer requires an explicit check against a known dataset. 
- Makes `RawDataLoader` a base class where some methods must be overridden 
- Adds `LSMSDataLoader` and `XYZDataLoader`
- Sets up example for QM9 dataset

Part of #99

### CODE REVIEW

**Comment by streeve** [comment]
Draft until we support multiple graph sizes

File: data_utils/raw_dataset_loader.py [hunk 1]
```diff
@@ -10,6 +11,23 @@
...
+    elif element == "C":
+        return 6
+    elif element == "N":
+        return 7
+    elif element == "O":
+        return 8
+    elif element == "F":
+        return 9
+    else:
+        print(element, ": Sorry Max & Pei")
```
Comment (pzhanggit) [inline data_utils/raw_dataset_loader.py:27]: Lol

Comment (pzhanggit) [reply data_utils/raw_dataset_loader.py:27]: Add Si, Fe, Cu etc.? They are already there in DFT data.

File: data_utils/raw_dataset_loader.py [hunk 2]
```diff
@@ -32,10 +61,14 @@ def load_raw_data(self, dataset_path: str, config):
         config: shows the target variables information, e.g, location and dimension, in data file
         """
 
-        node_feature_dim = config["node_features"]["dim"]
-        node_feature_col = config["node_features"]["column_index"]
-        graph_feature_dim = config["graph_features"]["dim"]
-        graph_feature_col = config["graph_features"]["column_index"]
+        self.node_position_dim = config["node_positions"]["dim"]
```
Comment (pzhanggit) [inline data_utils/raw_dataset_loader.py:64]: This feature is already in Main branch. Just realized I am reviewing a draft, will come back later.

**Comment by streeve** [comment]
Closing to be reopened in HydraGNN

**Pull request closed without merging**